packages/demo/bridge/support/stellar/main.go

```diff
@@ -0,0 +1,55 @@
+package stellar
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type MemoType int64
+
+const (
+	_ MemoType = iota
+	MemoTypeText
+	MemoTypeID
+	MemoTypeHash
+	MemoTypeReturn
+)
+
+func ToMemoType(memoType string) (MemoType, bool) {
+	switch memoType {
+	case "text":
+		return MemoTypeText, true
+	case "id":
+		return MemoTypeID, true
+	case "hash":
+		return MemoTypeHash, true
+	case "return":
+		return MemoTypeReturn, true
+	default:
+		return 0, false
+	}
+}
+
+func ExtractMemoToEthereumAddress(memoType, memoValue string) (ethAddress string, err error) {
+	switch memoType {
+	case "hash":
+		memoBytes, err := base64.StdEncoding.DecodeString(memoValue)
+		if err != nil {
+			return "", err
+		}
+		if len(memoBytes) != 32 {
+			return "", ErrInvalidEthAddress
```

Comment:
https://ethereum.stackexchange.com/questions/1374/how-can-i-check-if-an-ethereum-address-is-valid

does this cover the case of checksummed address?

Reply:
No, the `memoBytes` here is not a string with capitalization. Stellar `MEMO_TEXT` type memo field only allows a maximum of 28 bytes, we won't be able to fit a 40 character address string into it.